src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -262,7 +262,7 @@ object BoringUtils {
     val lcaResult = DataMirror.findLCAPaths(source, thisModule)
     if (lcaResult.isEmpty) {
       Builder.error(s"Cannot bore from $source to ${thisModule.name}, as they do not share a least common ancestor")
-      return DontCare
+      return DontCare.asInstanceOf[A]
```

Comment:
Is this not a runtime error?

Reply:
This one can be removed. I removed all the `DontCare`'s which are actually not necessary.